src/lib/components/BucketAggregation/BucketAggregationValues.js

```diff
@@ -75,13 +75,22 @@ class BucketAggregationValues extends Component {
       };
       const getChildAggCmps = (bucket) =>
         this.getChildAggCmps(bucket, selectedFilters);
+      let label = null;
+      if(valuesLabels){
+        label = _get(
+          valuesLabels.find(e => e.value === bucket.key  ),
+          'label',
+          bucket.key
+        );
+      }
       return (
         <ValueElement
           key={bucket.key}
           bucket={bucket}
           isSelected={isSelected}
           onFilterClicked={onFilterClicked}
           getChildAggCmps={getChildAggCmps}
+          valueLabel={label}
```

Comment:
minor: It is definitely not to be addressed now...but why don't we just name the prop `label`? The `value` prefix is unnecessary as the property refers to the label of the `ValueElement` component. I might miss something though :P But, again, not to be addressed now just curiosity :)

Reply:
I didn't want to confuse it with the whole aggregation label (the one you see on top of the filter)